### PR TITLE
Port SearchInput to NextJS

### LINKED ISSUE #1500: [next] Port SearchInput/ to NextJS
https://github.com/Seneca-CDOT/telescope/tree/master/src/frontend/gatsby/src/components/SearchInput
**What would you like to be added**:
Port all components in SearchInput/ to NextJS using TS
**Why would you like this to be added**:
Porting reason

### PR DESCRIPTION
**Pull request by HenryZerocool**
## Issue This PR Addresses

Fixes #1500 

## Type of Change

- [ ] **Bugfix**: Change which fixes an issue
- [ ] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [ ] **UI**: Change which improves UI
- [x] **Next**: Porting to NextJS

## Description

Port SearchInput to NextJS

## Checklist

- [ ] **Quality**: This PR builds and passes our npm test and works locally
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [ ] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

### CODE REVIEW

File: src/frontend/next/src/components/SearchInput/AuthorSearchInput.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import React from 'react';
```
Comment (c3ho) [inline src/frontend/next/src/components/SearchInput/AuthorSearchInput.tsx:2]: We don't need this line here

File: src/frontend/next/src/components/SearchInput/AuthorSearchInput.tsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+import PropTypes from 'prop-types';
+
+function AuthorSearchInput({ text, onChange, classes }: any) {
```
Comment (c3ho) [inline src/frontend/next/src/components/SearchInput/AuthorSearchInput.tsx:5]: If we're accepting props here, we should have an interface to type check everything. Check out the other PRs on how to do this.

It should also not return `any` as that is pretty hacky. Please take a look at documentation or examples on what a functional component in react for typesript.

File: src/frontend/next/src/components/SearchInput/AuthorSearchInput.tsx [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        aria-label="search telescope"
+        value={text}
+        onChange={onChange}
+      />
+      <datalist id="search-suggestions" />
+    </>
+  );
+}
+AuthorSearchInput.propTypes = {
+  text: PropTypes.string,
```
Comment (c3ho) [inline src/frontend/next/src/components/SearchInput/AuthorSearchInput.tsx:22]: we can probably get rid of checking propTypes here

**Review by c3ho — CHANGES_REQUESTED** [review]

**Comment by c3ho** [comment]
The other files other than the ones I commented will need the changes too. Btw the person reviewing should be the one to mark the issues resolved

File: src/frontend/next/src/components/SearchInput/AuthorSearchInput.tsx [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+interface SearchInput {
+  text: string;
+  onChange: () => void;
+  classes: {
+    input: string;
+  };
+}
+
+function AuthorSearchInput({ text, onChange, classes }: SearchInput) {
```
Comment (c3ho) [inline src/frontend/next/src/components/SearchInput/AuthorSearchInput.tsx:11]: Take a look at  https://github.com/Seneca-CDOT/telescope/pull/1485/files#diff-dfc72a93d99668af007bd9cfe8dbb5403cefb490f108b9ead155075124b4ef68R5-R7 to see how this part should be done

**Review by c3ho — CHANGES_REQUESTED** [review]

File: src/frontend/next/src/components/SearchInput/AuthorSearchInput.tsx (hunk 4 above)
Comment (HenryZerocool) [reply src/frontend/next/src/components/SearchInput/AuthorSearchInput.tsx:11]: Thanks for the instruction. I will update it

**Comment by tonyvugithub** [comment]
@HenryZerocool : Hi there, we are in the process of reviewing these migrated to Next PRs. Just wonder if you are still interested to work on this as we would like you to make some changes to your PR. Please let us know in case you decided to drop this then we can assign the issue to others. Thank you.

**Comment by PedroFonsecaDEV** [comment]
SearchInput PR: https://github.com/Seneca-CDOT/telescope/pull/1580

**Pull request closed without merging**